tests/BootstrapperTest.php

```diff
@@ -184,6 +187,70 @@
     expect($new_storage_path)->toEqual($expected_storage_path);
 });
 
+test('filesystem local storage has own public url', function() {
+    config([
+        'tenancy.bootstrappers' => [
+            FilesystemTenancyBootstrapper::class,
+        ],
+        'tenancy.filesystem.root_override.public' => '%storage_path%/app/public/',
+        'tenancy.filesystem.url_override.public' => 'public-%tenant_id%'
+    ]);
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+
+    tenancy()->initialize($tenant1);
+    $this->assertEquals(
+        'http://localhost/public-'.$tenant1->getTenantKey().'/',
+        Storage::disk('public')->url('')
+    );
+
+    tenancy()->initialize($tenant2);
+    $this->assertEquals(
+        'http://localhost/public-'.$tenant2->getTenantKey().'/',
+        Storage::disk('public')->url('')
+    );
```

Comment:
I think it'd be better if the tests actually tried to save some text file and then asserted that it can be fetched on the URL. Now we're only making assertions about the root URL, but we want to test that the feature actually works for files.

So **on top** of the root assertions, there should be assertions about files.

You can make that a separate test if you want. Or feel free to reorganize them in any way.

Reply:
I tried this, but it seems like the PR only takes care of generating the URL string – fetching the URL gives me a 404

Reply:
@stancl, [I changed the test](https://github.com/archtechx/tenancy/compare/b474ae219c844cde536a5ed0d052fc0b59e5cfaf..storage-url-conflict-resolution) according to our BC discussion